storage/pebble/registers_test.go

```diff
@@ -117,14 +165,60 @@ func Test_PayloadStorage_Versioning(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, expectedValue1ge3, value1)
 
-	value1, err = s.Get(height3+1, key1)
-	require.NoError(t, err)
-	require.Equal(t, expectedValue1ge3, value1)
+	// out of range
+	_, err = s.Get(height3+1, key1)
+	require.ErrorIs(t, err, storage.ErrHeightNotIndexed)
 
 	err = db.Close()
 	require.NoError(t, err)
 }
 
+func TestRegisters_LatestHeight(t *testing.T) {
+	t.Parallel()
+	cache := pebble.NewCache(1 << 20)
+	defer cache.Unref()
+	opts := DefaultPebbleOptions(cache, registers.NewMVCCComparer())
+
+	dbpath := path.Join(t.TempDir(), "versioning.db")
+	db, err := pebble.Open(dbpath, opts)
+	require.NoError(t, err)
+	// populate first height!
+	payload := make([]byte, 8)
+	initialHeight := binary.BigEndian.AppendUint64(payload, uint64(1))
+	err = db.Set(firstHeightKey(), initialHeight, nil)
+	require.NoError(t, err)
+	s, err := NewRegisters(db)
+	require.NoError(t, err)
+	require.NotNil(t, s)
+
+	// happy path
+	expected := uint64(1)
+	err = s.db.Set(latestHeightKey(), s.getEncodedHeight(expected), nil)
+	s.latestHeight = expected
+	require.NoError(t, err)
+	got, err := s.LatestHeight()
+	require.NoError(t, err)
+	require.Equal(t, got, expected)
+
+	// updating first height should not affect latest
+	firstHeight := uint64(0)
+	err = s.db.Set(firstHeightKey(), s.getEncodedHeight(firstHeight), nil)
```

Comment:
Ideally, we could create an import to import a empty checkpoint, so that the first height will be set by the importer without using this raw database calls.

The problem with the raw database calls is that if we ever refactor the database implementation, these tests would have to be updated as well.